### ISSUE TITLE
How to determine the atomic charge from the ados predicted by DP?

### ISSUE THREAD

**Issue by Wang-Zhiwei24** [Zhang-pchao/predict_atomic_charge#1]
I have two questions regarding the Predict-Atomic-Charge example using DP in this repository:

1. Relating DP-predicted DOS to atomic charges

- The model, developed in Phys. Rev. B 105, 174109, was used to predicts the Electronic Density of States (DOS). Could you clarify how to derive atomic charges (e.g. Hirshfeld-like charges) from these DOS predictions? 
- Is there a standard post-processing method/script provided in this example? Are there recommended references or acticles compatible with this strategy?

2. Preparing training data from CP2K output

-  If we use the cp2k software to calculation the Hirshfeld, how to build the training dataset (dos.npy) of DP from the cp2k out.log?
- Are there specific parsing scripts available for this conversion; 

Best wishes and looking forward to your reply

**Comment by Zhang-pchao** [comment]
# Hirshfeld Charge Model Training with CP2K and DeepMD

## Steps to Train the Charge Model

1. **Calculate Hirshfeld Charges in CP2K**  
   Add the following keyword in your CP2K input file to enable Hirshfeld charge calculation:  
   [Hirshfeld keyword](https://github.com/Zhang-pchao/OilWaterInterface/blob/main/DFT_Calculation/cp2k_input/m062x/m062x.inp#L145)

2. **Build the Training Dataset**  
   Extract the charge data (`atom_dos.npy`) from the `cp2k.out.log` file using this script:  
   [sp_cp2k2dp_charge.py](https://github.com/Zhang-pchao/predict_atomic_charge/blob/main/cp2k_Hirshfeld/sp_cp2k2dp_charge.py)

3. **Train the Charge Model**  
   Train the model following the settings in:  
   [run.json](https://github.com/Zhang-pchao/predict_atomic_charge/blob/main/model/run.json)

4. **More Information**  
   - For additional details, refer to the [DOS model manual](https://docs.deepmodeling.com/projects/deepmd/en/master/model/train-fitting-dos.html), the [Bohrium Notebook](https://bohrium.dp.tech/notebooks/6641686247) and the [paper](https://doi.org/10.1021/acs.langmuir.4c05004).  
   - The charge model is a simplified case of the DOS model, where the charge model is one-dimensional.

**Comment by Wang-Zhiwei24** [comment]
Thanks for your carefully reply, I have followed your steps you mentioned above to train the DP charge model. After I obtained this model, how should I use it to predict the atomic charges for the MD trajectories?

**Comment by Zhang-pchao** [comment]
After obtaining the model:

1. You can evaluate the model on a test set using the [dp test](https://bohrium.dp.tech/notebooks/6641686247) command. Note that you’ll first need to prepare the test data by modifying the [lammps_trj2dp_charge.py](https://github.com/Zhang-pchao/OilWaterInterface/blob/main/Analysis_Scripts/charge/lammps_trj2dp_charge.py) script accordingly.

2. Another way is to use the [ASE interface](https://github.com/AxelTG/BaH2/blob/main/dp_charges/bader_predict.py), which allows direct prediction of atomic charges from structure files.

**Comment by Wang-Zhiwei24** [comment]
Got it, thanks for sharing! I'll try it.
